File: apps/web/src/lib/db/schema/session_documents.ts
import {
    pgTable,
    uuid,
    integer, 
    text,
    timestamp,
    index,
} from "drizzle-orm/pg-core";
import { sessions } from "@/lib/db/schema/sessions";
import {
    fileTypeEnum,
} from "@/lib/db/enums";

export const session_documents = pgTable("session_documents",
{
    session_document_id : uuid().primaryKey().defaultRandom(),
    session_id : uuid()
        .references(() => sessions.session_id, {onDelete : "cascade"})
        .notNull(),
    file_name : text().notNull(),
    file_type : fileTypeEnum().notNull(),
    storage_url : text().notNull(),
    extracted_text : text(),
    token_count : integer(),
    uploaded_at : timestamp({
            withTimezone : true
        })
        .notNull()
        .defaultNow(),

    created_at : timestamp({
        withTimezone : true
    })
    .defaultNow()
    .notNull(),

    updated_at : timestamp({
        withTimezone : true
    })
    .notNull()
    .defaultNow(),
},
    (table) => [
        index("session_documents_session_id_idx")
        .on(table.session_id),
    ]
);

File: apps/ai-service/models/orm/session_documents.py
from datetime import datetime
import uuid

from sqlalchemy import (
    DateTime,
    ForeignKey,
    Index,
    Integer,
    Text,
    func,
)
from sqlalchemy.dialects.postgresql import UUID as PG_UUID
from sqlalchemy.orm import Mapped, mapped_column

from models.orm.base import (
    Base,
    pg_value_enum,
)
from models import FileType


class SessionDocument(Base):
    __tablename__ = "session_documents"

    __table_args__ = (
        Index(
            "session_documents_session_id_idx",
            "session_id",
        ),
    )

    session_document_id: Mapped[uuid.UUID] = mapped_column(
        PG_UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4,
    )

    session_id: Mapped[uuid.UUID] = mapped_column(
        PG_UUID(as_uuid=True),
        ForeignKey(
            "sessions.session_id",
            ondelete="CASCADE",
        ),
        nullable=False,
    )

    file_name: Mapped[str] = mapped_column(nullable=False)

    file_type: Mapped[FileType] = mapped_column(
        pg_value_enum(
            FileType,
            pg_name="file_type",
        ),
        nullable=False,
    )

    storage_url: Mapped[str] = mapped_column(
        nullable=False,
    )

    extracted_text: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    token_count: Mapped[int | None] = mapped_column(
        Integer,
        nullable=True,
    )

    uploaded_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )